mautic/mautic
# Contacts should export via mail instead of immediate download

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | ❌ 
| New feature/enhancement? (use the a.x branch)      | ✅ 
| Deprecations?                          | ❌ 
| BC breaks? (use the c.x branch)        | ✅ 
| Automated tests included?              | ✅  
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:



This PR puts active download of contact exports (CSV only) to be queued, and email link is sent over email to download the exports. This enables users to download large contacts without hampering the UX like longer times to get the export file and freezing the page etc. Excel export will work as it is.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Run the migration using `ddev exec bin/console doctrine:migrations:execute --up 20220429091934`
3. Go to contacts page
4. Click on export CSV from the menus (Try different filters like ids, emails etc)
5. Check notification if appropriate message is displayed
6. Run `php bin/console mautic:contacts:scheduled_export --ids=<comma separated contact_export_scheduler ids>` (ids are optional)
7. Once email is received, click on link to download the expected contact export file and also check notification for correct message
8. Verify if data in the file is correct
9. Remove the migration if required using `ddev exec bin/console doctrine:migrations:execute --down 20220429091934`




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11272"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>



---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203523494025843
